packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js

```diff
@@ -0,0 +1,187 @@
+import Model, { attr } from '@ember-data/model';
+import Store from '@ember-data/store';
+import JSONAPIAdapter from '@ember-data/adapter/json-api';
+import EmberObject from '@ember/object';
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { resolve } from 'rsvp';
+
+class Person extends Model {
+  @attr
+  firstName;
+
+  @attr
+  lastName;
+
+  // override to not call serialize()
+  toJSON() {
+    const { id, firstName, lastName } = this;
+    return {
+      id,
+      type: this.constructor.modelName,
+      attributes: {
+        firstName,
+        lastName,
+      },
+    };
+  }
+}
+
+module('inteegration/delete-record - running deleteRecord with minimum serializer', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function(assert) {
+    this.owner.register('service:store', Store);
+    this.owner.register('model:person', Person);
+  });
+
+  test('save after deleting record does not call normalizeResponse and serialize', async function(assert) {
+    let normalizeResponseCalled = 0;
+    let _payloads = [
+      {
+        id: '1',
+        type: 'person',
+        attributes: {
+          firstName: 'John',
+          lastName: 'Smith',
+        },
+      },
+      '',
```

Comment:
if we leave a payload here for the `deleteRecord` response we do in fact call `normalizeResponse`. Let's add a test for that as well.

I added an issue for discussing the fact we don't always call `normalizeResponse` here: https://github.com/emberjs/data/issues/6541

I was not aware of this and it seems a bit bonkers.

Reply:
I am confused about how it makes any sort of sense to have a response to a DELETE request and why we should do anything with it if there is. JSON-API I believe requires a 204 No Response.

Reply:
Added additional test.